internal/datacoord/meta.go

```diff
@@ -800,11 +800,11 @@ func (m *meta) CompleteMergeCompaction(compactionLogs []*datapb.CompactionSegmen
 
 	var startPosition, dmlPosition *internalpb.MsgPosition
 	for _, s := range segments {
-		if dmlPosition == nil || s.GetDmlPosition().Timestamp < dmlPosition.Timestamp {
+		if dmlPosition == nil || s.GetDmlPosition().GetTimestamp() < dmlPosition.GetTimestamp() {
 			dmlPosition = s.GetDmlPosition()
 		}
 
-		if startPosition == nil || s.GetStartPosition().Timestamp < startPosition.Timestamp {
+		if startPosition == nil || s.GetStartPosition().GetTimestamp() < startPosition.GetTimestamp() {
```

Comment:
maybe we shall check whether `s.GetStartPostion()` and `s.GetDmlPosition()` is nil

Reply:
That should be OK, 0 will be returned in `s.GetStartPostion()` or `s.GetDmlPosition()` is nil:

```
func (m *MsgPosition) GetTimestamp() uint64 {
	if m != nil {
		return m.Timestamp
	}
	return 0
}
```

Reply:
If the segment startPosition or dmlPosition is nil, using GetStartPosition().GetTimeStamp() shall always less or equal to current minimal postion, which means a nil value will overwrite the minimal position.

Reply:
Sure that makes sense.